### ISSUE TITLE
Animation disappears on using withSpring

### ISSUE THREAD

**Issue by AhmadNadeemSiddiqui** [software-mansion/react-native-reanimated#4554]
### Description

I have applied some animation so that I get a shaky ui effect when user fails validation. But it disappears the animation.

```
const inputTransformValue = useSharedValue(0);

 const shakeUI = () => {
    inputTransformValue.value = withSequence(
      withTiming(-10, {duration: 2000}),
      withSpring(0, {
        damping: 8,
        mass: 0.5,
        stiffness: 1000,
        restDisplacementThreshold: 0.1,
      }),
    );
  };

  const inputStyle = useAnimatedStyle(() => {
    return {
      transform: [{translateX: inputTransformValue.value}],
    };
  });
```

https://github.com/software-mansion/react-native-reanimated/assets/121891680/4d331219-0135-4b11-908e-3a6d71ef061e

### Steps to reproduce

```
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withSequence,
  withSpring,
  withTiming,
}
```

### Snack or a link to a repository

[[www.github.com](https://github.com/AhmadNadeemS/podify)](https://github.com/AhmadNadeemS/podify)

### Reanimated version

3.2.0

### React Native version

0.71.4

### Platforms

Android

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by ammarahm-ed** [comment]
Having a similar issue.

**Comment by Pnlvfx** [comment]
Same

**Comment by Pnlvfx** [comment]
Same. It was working on 3.1.0 and disappear on 3.2.0

**Comment by efstathiosntonas** [comment]
There's an open [PR](https://github.com/software-mansion/react-native-reanimated/pull/4512) for this

**Comment by Pnlvfx** [comment]
> There's an open [PR](https://github.com/software-mansion/react-native-reanimated/pull/4512) for this

Yeah, I’m waiting for it to upgrade

**Comment by tomekzaw** [comment]
Hey @AhmadNadeemSiddiqui, thanks for reporting this problem. As @efstathiosntonas mentioned, it should be fixed with #4512, and since this PR just got merged, I'm closing this issue. Please let us know if it works for you.
